src/index.tsx

```diff
@@ -125,8 +131,37 @@ async function activatePlugin(
 ): Promise<void> {
   const { commands } = app;
   const trans = translator.load('jupyterlab');
-  const fileBrowserTracker = browserFactory.tracker;
   const api = new SchedulerService({});
+
+  // Try calling an API to verify that the server extension is actually installed
+  let serverExtensionOk = false;
+  api
+    .getJobs({ max_items: 1 })
+    .then(response => {
+      serverExtensionOk = true;
+    })
+    .catch((responseError: ServerConnection.ResponseError) => {
+      // Response error: got something other than 200 OK in response
+      const responseCode = responseError.response.status;
+      // Treat a 404 Not Found response as the backend not being present.
+      if (responseCode === 404) {
+        showDialog({
+          title: trans.__('Jupyter Scheduler server extension not found'),
+          body: SERVER_EXTENSION_404,
+          buttons: [Dialog.okButton()]
+        }).catch(console.warn);
+      } else {
+        // Treat other response codes as transient errors; optimistically hope that
+        // server extension is present
```

Comment:
Should we handle 5xx errors here as well? I'm personally neutral to it myself, but would like to start a brief dialogue on this.

Reply:
I'm going to consider only a 404 error as a hard "no' for now; willing to revisit this in a future issue

Reply:
Sounds good.